### PR TITLE
Start es6 + babel + webpack conversion

### PR DESCRIPTION
**Pull request by weikinhuang**
There will be a series of PRs to follow to modernize this repo

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -1,105 +0,0 @@
-/**
- * Graph
- * - head: Head root node
- * - traversals: Map of all traversed pathes; keyed hash => uri
- * - nodes: Map of all known node; keyed uri => Node
- */
-define(['promise', './xhr', './resource'], function(Promise, xhr, Resource) {
```
Comment (weikinhuang) [inline lib/client.js]: I didn't change anything in the files except convert from amd to import

Comment (weikinhuang) [inline lib/client.js]: No point in checking the bodies of these files

File: index.js [hunk 2]
```diff
@@ -1,9 +1 @@
-define([
-  './lib/client'
-], function(Client) {
-  'use strict';
-
-  return {
-    Client: Client
-  };
-});
+module.exports = { Client: require('./src/client') };
```
Comment (mikesherov) [inline index.js]: Why not es6 modules?

Comment (weikinhuang) [inline index.js]: no reason

File: karma.conf.js [hunk 3]
```diff
@@ -8,6 +8,16 @@ var webpackConfig = require('./webpack.config.js');
 // delete webpackConfig.entry;
 webpackConfig.devtool = 'eval';
 
+// Only run coverage report during `npm test`
+if (process.env.npm_lifecycle_event === 'test') {
```
Comment (kaicataldo) [inline karma.conf.js]: This is cool :+1: Didn't know about this

File: index.js (hunk 2 above)
Comment (weikinhuang) [inline index.js]: Whoops, actually there was a reason, export from is in stage-1
`export { default as Client } from './src/client';`

File: karma.conf.js [hunk 4]
```diff
@@ -4,8 +4,62 @@
 
 'use strict';
 
-var commonConfig = require('./karma-common-config');
+var webpackConfig = require('./webpack.config.js');
+// delete webpackConfig.entry;
```
Comment (kaicataldo) [inline karma.conf.js]: Curious about the commented out stuff

File: karma.conf.js [hunk 5]
```diff
@@ -4,8 +4,62 @@
...
-  config.set(commonConfig);
+  config.set({
+    basePath: '',
+
+    frameworks: ['mocha', 'chai', 'sinon-chai'],
+
+    files: [
+      'node_modules/babel-polyfill/dist/polyfill.js',
+      require.resolve('whatwg-fetch'),
+      // 'node_modules/chai-as-promised/lib/chai-as-promised.js',
```
Comment (kaicataldo) [inline karma.conf.js]: Curious about the commented out stuff

File: karma.conf.js (hunk 3 above)
Comment (weikinhuang) [inline karma.conf.js]: Actually this is in all the behance repos, I would prefer to always run coverage, but not output the html version all the time

File: test/specs/client.js [hunk 6]
```diff
@@ -1,4 +1,5 @@
-define(['chai-as-promised', 'lib/client', 'lib/resource'], function(chaiAsPromised, Client, Resource) {
+
```
Comment (kaicataldo) [inline test/specs/client.js]: Extra line

File: test/specs/client.js [hunk 7]
```diff
@@ -1,4 +1,5 @@
...
+define(['chai-as-promised', 'src/client', 'src/resource'], function(chaiAsPromised, Client, Resource) {
```
Comment (kaicataldo) [inline test/specs/client.js]: Is there a reason we're not using ES6 modules?

File: karma.conf.js (hunk 4 above)
Comment (weikinhuang) [inline karma.conf.js]: debugging

File: karma.conf.js (hunk 5 above)
Comment (weikinhuang) [inline karma.conf.js]: this line is still WIP

File: test/specs/client.js (hunk 6 above)
Comment (weikinhuang) [inline test/specs/client.js]: whoops

File: test/specs/client.js (hunk 7 above)
Comment (kaicataldo) [inline test/specs/client.js]: Whoops, didn't see the comments because I was going through commit by commit. Looks like you answered this already.

Comment (weikinhuang) [inline test/specs/client.js]: this is going to be in the next pr

**Comment by kaicataldo** [comment]
:+1:

File: karma.conf.js (hunk 5 above)
Comment (kaicataldo) [inline karma.conf.js]: Facemailed

File: index.js (hunk 2 above)
Comment (mikesherov) [inline index.js]: Would then prefer an import and export statement over the mixed metaphor of different import/export styles.

Comment (weikinhuang) [inline index.js]: sure thing!

File: karma.conf.js (hunk 5 above)
Comment (designjockey) [inline karma.conf.js]: Consider leaving a todo comment

File: src/xhr.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import Uri from './uri';
+
+function isCrossOrigin(origin, remote) {
+  remote = Uri.parse(remote);
+
+  return remote.isAbsolute() &&
+         !(origin.host === remote.host && origin.port === remote.port);
```
Comment (designjockey) [inline src/xhr.js]: This is probably better off in the same line as the return or consider creating a var for the condition.

**Comment by designjockey** [comment]
👍

File: src/xhr.js (hunk 8 above)
Comment (weikinhuang) [inline src/xhr.js]: I only changed the whitespace in everything under the import statement

**Pull request merged**